### PR TITLE
SONARIAC-1529 Secondary locations on other files should be disabled with a specific option per rule

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

**Review by mstachniuk — CHANGES_REQUESTED** [review]
I would like to have the internal property for this feature. This will help us validate the impact on Peachee much more easily. 
Please also check if the feature should be disabled for all K8s/Helm rules as they may also potentially break the analysis.

**Comment by rudy-regazzoni-sonarsource** [comment]
As indicated, I updated the Jira ticket and modified the PR: now we can set an option per rule to remove secondary location on other files.

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheckTest.java [hunk 1]
```diff
@@ -76,4 +76,15 @@ void testLinkedAccountNonCompliantChartWithShiftedLocation() {
     KubernetesVerifier.verify("AutomountServiceAccountTokenCheck/LinkedAccount/NonCompliant/SensitiveValueChart/templates/automount_service_account_token_pod_linked.yaml", check,
       List.of(expectedIssue));
   }
+
+  @Test
+  void shouldRaiseSecondaryLocationOnLinkedAccount() {
+    String root = "AutomountServiceAccountTokenCheck/LinkedAccount/NonCompliant/SensitiveValue/";
+    var secondaryLocation1 = new SecondaryLocation(TextRanges.range(10, 22, 10, 45), "Through this service account");
+    var secondaryLocation2 = new SecondaryLocation(TextRanges.range(6, 30, 6, 34), "Change this setting",
+      "AutomountServiceAccountTokenCheck/LinkedAccount/NonCompliant/SensitiveValue/linked_account_service_token.yaml");
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/AutomountServiceAccountTokenCheckTest.java:85]:
```suggestion
      root + "linked_account_service_token.yaml");
```

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitorTest.java [hunk 2]
```diff
@@ -174,6 +174,18 @@ void shouldReportWithSecondaryLocationOnDifferentFile() {
...
+  @Test
+  void shouldDiscardSecondaryLocationOnDifferentFileWhenDisabled() {
+    var inputFileContext = createInputFileContextMock(false);
+    when(inputFileContext.sensorContext.config().getBoolean(KubernetesChecksVisitor.DISABLE_SECONDARY_LOCATIONS_IN_OTHER_YAML_KEY + ".testRule")).thenReturn(Optional.of(true));
+    TextRange range = range(1, 0, 1, 1);
+    var secondaryLocationSameFile = new SecondaryLocation(range(1, 0, 2, 3), "testIssueSecondary");
+    var secondaryLocationOtherFile = new SecondaryLocation(range(1, 0, 2, 3), "testIssueSecondary", "my/other/file.yaml");
+    var customVisitor = prepareVisitorToRaise("testIssue", List.of(secondaryLocationSameFile, secondaryLocationOtherFile));
+    customVisitor.scan(inputFileContext, tree);
+    verify(inputFileContext, times(1)).reportIssue(RuleKey.of("testRepo", "testRule"), range, "testIssue", List.of(secondaryLocationSameFile));
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitorTest.java:186]: Nitpik. The `verify(inputFileContext)` is an equivalent of `verify(inputFileContext, times(1))`. 
I already fixed it in my PR https://github.com/SonarSource/sonar-iac/pull/1411 , so please leave it for now.

**Review by mstachniuk — APPROVED** [review]
LGTM

**Pull request merged**